src/libraries/Common/src/System/Net/Http/WinInetProxyHelper.cs

```diff
@@ -117,13 +117,25 @@ public WinInetProxyHelper()
             //    fAutoLogonIfChallenged member set to TRUE.
             //
             // We match behavior of WINHTTP_ACCESS_TYPE_AUTOMATIC_PROXY and ignore errors.
+
+            string destination = uri.AbsoluteUri;
+            // Underlying code does not understand WebSockets so we need to convert it to http or https.
+            if (uri.Scheme == UriScheme.Wss)
+            {
+                destination = "https" + destination.Substring(3);
```

Comment:
nit: can we do UriScheme.Wss.Length here instead of hardcoding "3"?

Similarly below.

Reply:
And, we should probably use UriScheme.Http and UriScheme.Https rather than hardcoded string values.

Reply:
updated.